keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/ClusterExportHelper.kt

```diff
@@ -0,0 +1,110 @@
+ */
+@Component
+class ClusterExportHelper(
+  private val cloudDriverService: CloudDriverService,
+  private val orcaService: OrcaService
+) {
+  val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  /**
+   * Retrieve server group entity tags from clouddriver to identity the "source" of the server group (pipeline or
+   * orchestration), then retrieve execution details from orca to determine the deployment strategy used for that
+   * server group.
+   */
+  suspend fun discoverDeploymentStrategy(
+    cloudProvider: String,
+    account: String,
+    application: String,
+    serverGroupName: String
+  ): ClusterDeployStrategy? {
+    return kotlinx.coroutines.coroutineScope {
+      val entityTags = async {
+        log.debug("Looking for entity tags on server group $serverGroupName in application $application, " +
+          "account $account in search of pipeline/task correlation.")
+        cloudDriverService.getEntityTags(
+          cloudProvider = "aws",
+          account = account,
+          application = application,
+          entityType = "servergroup",
+          entityId = serverGroupName
+        )
+      }.await()
+
+      if (entityTags.isEmpty()) {
+        log.warn("Unable to find entity tags for server group $serverGroupName in application $application, " +
+          "account $account.")
+        return@coroutineScope null
+      }
+
+      val spinnakerMetadata = entityTags.first().tags
+        .find { it.name == "spinnaker:metadata" }
+        ?.let { it.value as? Map<*, *> }
+
+      if (spinnakerMetadata == null ||
+        spinnakerMetadata["executionType"] == null ||
+        spinnakerMetadata["executionId"] == null
+      ) {
+        log.warn("Unable to find Spinnaker metadata for server group $serverGroupName in application $application, " +
+          "account $account in entity tags.")
+        return@coroutineScope null
+      }
+
+      val executionType = spinnakerMetadata["executionType"]!!.toString()
+      val executionId = spinnakerMetadata["executionId"]!!.toString()
+      val execution = async {
+        when (executionType) {
+          "orchestration" -> orcaService.getOrchestrationExecution(executionId)
+          "pipeline" -> orcaService.getPipelineExecution(executionId)
+          else -> throw SystemException("Unsupported execution type $executionType in Spinnaker metadata.")
+        }
+      }.await()
+
+      val context = if (executionType == "pipeline") {
+        // TODO: or clone?
+        execution.getDeployStageContext()
+      } else { // orchestration (i.e. a task)
+        // TODO: or cloneServerGroup?
+        execution.getTaskContext("createServerGroup")
+      }
+
+      when (val strategy = context?.get("strategy")) {
+        "redblack" -> RedBlack.fromOrcaStageContext(context)
+        "highlander" -> Highlander
+        null -> throw InvalidRequestException("Deployment strategy information not found for server group $serverGroupName " +
```

Comment:
same comment re: exceptions here. Is it worth throwing them and borking the export when we're just trying to be extra helpful? I think a log message might be better